docs/guides/concurrency.md

```diff
@@ -250,13 +253,14 @@ Output:
 ```
 Before first receive
 Before first send
+Before second send
 1
 Before second receive
-Before second send
 2
+
 ```
 
-Note that when the program executes a `receive`, that fiber blocks and execution continues with the other fiber. When `send` is executed, execution continues with the fiber that was waiting on that channel.
+Note that when the program executes a `receive`, that fiber blocks and execution continues with the other fiber. When `channel.send(1)` is executed, execution continues because non-blocked, until `channel.send(2)`, because channel blocked, execution continues with the fiber that was waiting on that channel.
```

Comment:
```suggestion
Note that when the program executes a `receive`, the current fiber blocks and execution continues with the other fiber. When `channel.send(1)` is executed, execution continues because `send` is non-blocking if the channel is not yet full. However, `channel.send(2)` does cause the fiber to block because the channel (which has a size of 1 by default) is full, so execution continues with the fiber that was waiting on that channel.
```